contents/testing/integration-testing.md

```diff
@@ -0,0 +1,149 @@
+
+#### [Big bang approach](https://www.tutorialspoint.com/software_testing_dictionary/big_bang_testing.htm): 
+
+This approach involves integrating all the components in your design diagram together and testing everything at once in a complete state. This is convenient but it is difficult to isolate defects and there is a high chance of missing critical underlying defects. Big bang integration testing is usually used for smaller applications with few components. 
+
+An example of how big bang integration testing can be applied to our given example is shown below:
+
+![Big bang integration testing diagram](integration-test/big-bang-integration.png "Big bang integration testing diagram]")
+
+#### [Incremental testing approach](https://www.quora.com/What-is-incremental-testing-in-software):
+
+This approach involves integrating two or more logically related components. The other related components are added and tested for proper functioning. This is repeated until all the components are joined and tested successfully. It is usually preferred with any applications having more than 10 different components.
+
+Incremental integration testing is further split into the 3 approaches shown below:
+
+| | [Top-down approach](https://www.guru99.com/integration-testing.html#9) | [Bottom-up approach](https://www.guru99.com/integration-testing.html#8) | [Hybrid / sandwich approach](https://www.guru99.com/integration-testing.html#10) |
+| -- | -- | -- | -- |
+| **Description**| High level modules are tested first and then lower level modules until all the modules are tested. | The reverse of top-down approach. | A mix of both top-down bottom-up approaches. |
+| <span style="color:red">**Advantages**</span>| Early discovery of high level architecture / design defects | Easier to create test cases bottom up | Beneficial for big project to distribute tasks on testing
+| | Main control points of the system are tested early | Critical modules on functionalities ae tested first | Allow top-down and bottom-up approach to run side by side | 
+| <span style="color:blue">**Disadvantages**</span> | Significant low level modules are tested late in the cycle | There is no testable working system until the higher level modules are build | It is difficult to test for highly interconnected modules |
+| | A [<tooltip content="A program that simulates the behaviours of software components">stub</tooltip>](https://stackoverflow.com/questions/463278/what-is-a-stub) is not perfect to simulate data flow | A [<tooltip content="Module with dummy code to temporalities replace a module">driver</tooltip>](http://www.professionalqa.com/test-driver) test is even harder to write than stub | Higher cost from using both driver and stub. You can understand the difference [here](https://www.quora.com/What-is-the-difference-between-stubs-and-drivers-in-software-testing)
```

Comment:
Why is a stub not perfect to simulate data flow? (Maybe elaborate that stub only contains very basic, hardcoded, simulated data)